src/player_activity.h

```diff
@@ -41,6 +41,8 @@ class player_activity
         int position;
         /** An activity specific value. */
         std::string name;
+        std::string adds_fault;
+        std::string flags;
```

Comment:
str_value adds_fault;
        str_value flags;

??

Reply:
I mean you can use `player_activity::str_values` to store activity-specific values. But since I'm going to change the activity code to implement multiple mending actions for a single fault, I'd like you to hold off these changes until I finish my implementation.

Reply:
OK so I won't change anything for now and then you're done let me know I will resolve conflicts and then it should be good to merge.

Reply:
I've posted the infrastructure for multiple mending methods in #35279. After it's merged you can  implement both cleaning methods for `fault_gun_dirt` using the new format.

Reply:
> I've posted the infrastructure for multiple mending methods in #35279. After it's merged you can implement both cleaning methods for `fault_gun_dirt` using the new format.

Is this how you wanted two different methods of fixing the same fault to look?
![image](https://user-images.githubusercontent.com/45959491/68446443-aefb6300-01aa-11ea-9901-7570d07099c9.png)

Reply:
Yes, exactly.

Reply:
```suggestion
```

Reply:
> Yes, exactly.

I thought the issue was the list should list not solutions, but faults. Was expecting the new layout to list "blackpowder fouling" and then when you click on it another popup would come up with "clean" and "clean and lube." :confused:

Reply:
> I thought the issue was the list should list not solutions, but faults

Maybe I wasn't being clear, but no, that wasn't the issue. The propose of my change was to keep the fault names unchanged in the item info display, and to avoid hacking by properly supporting different mending methods.

> list "blackpowder fouling" and then when you click on it another popup would come up with "clean" and "clean and lube."

I think listing the mending methods together makes the menu more readable. Showing another popup to ask for the mending methods only increases the number of keypresses to mend a fault. Maybe it will be preferred to show a second popup if the number of mending methods significantly increases in the future, but I doubt that will happen any time soon.